### PR TITLE
Make dwds more conformant to vm_service protocol

### PR DESCRIPTION
**Pull request by annagrin**
See: https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#getstack 

- Throw `SentinelException` if the isolate id is unrecognized
    vm service protocol requires all isolate-related API to return `Sentinel` if the isolate
    had exited, which translates to throwing `SentinelException` in the implementation.
    Make dwds conform to that spec.

 - Throw `RPCError` if `getStack` is called when not paused
    vm service protocol requires getStack to always return a non-null value or a 
    sentinel, but dwds is limited by chrome devtools API that do not curently give a
    way to get the stack while running. Throw RPC error to indicate that something
    unexpected has happened.

Helps: https://github.com/dart-lang/webdev/issues/1369

 - Update tests using `expect` on Futures to use `expectLater` instead. 
   This makes sure the current test is finished before the next one starts, and 
   reduces possibilities of flakes.
 - Fix race conditions in chrome_proxy_tests and enable skipped test
 - Update tests to expect errors on `getStack` called while running.

Closes: https://github.com/dart-lang/webdev/issues/721

### CODE REVIEW

File: dwds/lib/src/debugging/debugger.dart [hunk 1]
```diff
@@ -9,7 +9,7 @@ import 'dart:math' as math;
 
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart';
-import 'package:vm_service/vm_service.dart' as vm_service;
+//import 'package:vm_service/vm_service.dart' as vm_service;
```
Comment (bkonyi) [inline dwds/lib/src/debugging/debugger.dart:12]: Guessing you didn't mean to leave this commented out? :-)

File: dwds/test/chrome_proxy_service_test.dart [hunk 2]
```diff
@@ -804,25 +809,28 @@ void main() {
...
-        return stack;
+        Breakpoint bp;
+        try {
+          bp = await service.addBreakpoint(isolateId, mainScript.id, line);
+          // Wait for breakpoint to trigger.
+          await stream
+              .firstWhere((event) => event.kind == EventKind.kPauseBreakpoint);
+          return await service.getStack(isolateId, limit: limit);
+        } finally {
+          // Remove breakpoint amd resume so it doesn't impact other tests.
```
Comment (bkonyi) [inline dwds/test/chrome_proxy_service_test.dart:828]: "Remove breakpoint _and_ resume"

**Review by bkonyi — APPROVED** [review]
LGTM with nits. Thanks for the quick fix!

File: dwds/lib/src/debugging/debugger.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart:12]: ah thanks!

File: dwds/test/chrome_proxy_service_test.dart (hunk 2 above)
Comment (annagrin) [reply dwds/test/chrome_proxy_service_test.dart]: thanks!

Comment (annagrin) [reply dwds/test/chrome_proxy_service_test.dart]: done.

File: dwds/lib/src/debugging/debugger.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart]: done.

**Pull request merged**